contrib/IECoreUSD/src/IECoreUSD/VolumeAlgo.cpp

```diff
@@ -96,9 +98,8 @@ IECore::ObjectPtr readVolume( pxr::UsdVolVolume &volume, pxr::UsdTimeCode time,
 			continue;
 		}
 
-		SdfAssetPath fieldAssetPath;
-		fieldAsset.GetFilePathAttr().Get( &fieldAssetPath, time );
-		const std::string fieldFileName = fieldAssetPath.GetResolvedPath();
+		ConstDataPtr fieldFileNameData = DataAlgo::fromUSD( fieldAsset.GetFilePathAttr(), time );
+		const std::string fieldFileName = runTimeCast<const StringData>( fieldFileNameData )->readable();
```

Comment:
`runTimeCast` should always be accompanied with a check for `null` - by using `runTimeCast` you are saying that you don't know what the type is. I think it would be reasonable to use `static_cast` or `assertStaticCast` here though, since we know the result type.

Reply:
Ah, I was not quite clear on when to use `runTimeCast` vs `static_cast`, thanks for that explanation. I went with `static_cast` in https://github.com/ImageEngine/cortex/pull/1426/commits/ffe45071a4b37022e22bd37ca7601c8815aeee0a since `dataFromSdfAssetPath()` always returns at least an empty `StringData`.